### PR TITLE
Todo Application with ReactJS

### PR DESCRIPTION
**Pull request by dewslyse**
In this PR, I built a Todo app using `React.js`. Tasks can be added using the input form. Added tasks can be marked as completed upon completion and they can also be deleted.

### CODE REVIEW

**Comment by isaacpitwa** [comment]
**Hi @dewslyse 👋🏿** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there 🏅 

## Highlights
- Your repo is professional. README file is descriptive ✔️  Commit messages should be meaningful.✔️
-  All linters Pass  ✔️ 
- Gitflow was correctly used ✔️
- Application  works as expected ✔️
- All concepts that have been learned so far were implemented correctly ✔️ 
- Impressing  quality of the code ✔️

## Required Changes ♻️
 [Optional] - You could try to be specific in the expected number prop types i.e  specifying the exact type using `Proptype.ofArray(PRoptype.ofObject())` on [line](https://github.com/dewslyse/mv-react-todo-app/blob/d90c236fb9b108e8221fb566b1c7faa8b9b09c01/src/components/TodosList.js#L24)



_Check the comments under the review._




almost-there

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @isaacpitwa.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/TodoItem.js [hunk 1]
```diff
@@ -3,16 +3,24 @@ import PropTypes from 'prop-types';
 
 class TodoItem extends PureComponent {
```
Comment (isaacpitwa) [inline src/components/TodoItem.js:5]: Using Class components with PureComponents risks your component to risk **prop changes will be ignored If it couldn't find a difference in shallow comparison**  I am sure your using it for the right reason 😊 performance 🚀

File: src/components/TodosList.js [hunk 2]
```diff
@@ -9,11 +9,15 @@ import TodoItem from './TodoItem';
 
 class TodoList extends PureComponent {
```
Comment (isaacpitwa) [inline src/components/TodosList.js:5]: I have noticed the use of PureCompoent for class components. These risks **prop changes will be ignored If it couldn't find a difference in shallow comparison**. I am sure you use it for the right reasons 😊. Performance 🚀🚀

File: package.json [hunk 3]
```diff
@@ -18,7 +20,9 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d build"
```
Comment (isaacpitwa) [inline package.json:25]: [Optional] You can choose to combine pre-deploy script with deploy scripts with one command. using 
 ``` 
"deploy": "npm run build && gh-pages -d build"
```
Fasten your deployment time 🚀 😃

File: src/components/TodoItem.js (hunk 1 above)
Comment (dewslyse) [reply src/components/TodoItem.js:5]: @isaacpitwa , thanks for the review 😎

File: package.json (hunk 3 above)
Comment (dewslyse) [reply package.json:25]: @isaacpitwa, I didn't know that. Thank you very much 🙏🏽

**Review by isaacpitwa — APPROVED** [review]
Hi  @dewslyse  👋🏽
You are all set, there is nothing else to add 🥳.
Go ahead and submit your project.

**Pull request merged**